syne_tune/tuner.py

```diff
@@ -99,11 +101,28 @@ def __init__(
 
         # inform the backend to the folder of the Tuner. This allows the local backend
         # to store the logs and tuner results in the same folder.
-        self.backend.set_path(results_root=self.tuner_path)
+        self.backend.set_path(results_root=str(self.tuner_path))
         self.callbacks = callbacks if callbacks is not None else [self._default_callback()]
 
         self.tuning_status = None
 
+    def _check_synchronous_scheduler(
```

Comment:
Can we move this logic outside of the Tuner? (for instance in the caller or in a factory)

It seems to me that the caller or a factory could make this assert and type switch case and that it would be a better place since this logic is highly specific to one case.

Reply:
Well, factories already take this into account, but then we tell people to write launcher scripts, in which scheduler and Tuner are created "by hand" ...

Reply:
OK, the alternative here is to display a different error message in SynchronousHyperband